File: juego_del_ahorcado.py
import random


def read():
    palabrasArray = []
    with open("./archivos/palabras.txt", "r", encoding="utf-8") as f:
        for palabra in f:
            palabrasArray.append(palabra.strip('\n'))
    
    rangoLimite = len(palabrasArray) - 2

    numeroRamdom = random.randint(0,rangoLimite)

    palabraSelec = palabrasArray[numeroRamdom]

    return palabraSelec

def compararPalabra(palabra):

    palabraArray = []
    palabraNueva = []

    for letra in palabra:
        palabraArray.append(letra)
    
    print(f'La palabra tiene {len(palabraArray)} digitos ...\n')
    print(palabraArray)


    for oportunidad in range(0,10):

        try:
            letra = input('Digite una letra: ')
            index = palabraArray.index(letra)
            palabraNueva.insert(index, letra)

            print(palabraNueva)

            if (palabraNueva == palabraArray):
                print('Ganaste!! ...')
                break

        except ValueError:
            print('No existe la letra en la palabra ...')

    print('Perdiste!!! No adivinaste la palabra ...')
    

def run():
    print('##### JUEGO DEL AHORCADO #####')
    palabraSelect = read()
    compararPalabra(palabraSelect)

    
    




if __name__ == '__main__':
    run()

File: juego_del_ahorcado.js
const inquirer = require('inquirer');
const fs = require('fs');
require('colors');


const menu = async (palabraBD) => {

    const palabraArrayFiltrada = palabraBD.split('').filter(item => item != '\r');
    const limite = 15;
    const palabraGenerada = [];

    console.log('###### JUEGO DEL AHORCADO ###### \n');

    console.log(`La palabra tiene ${palabraArrayFiltrada.length} caracteres ...\n`);

    const pregunta = [
        {
            type: 'input',
            name: 'letra',
            message: 'Digite una letra: '
        }
    ]

    for (let i=0; i<=limite; i++){

        if (palabraGenerada === palabraArrayFiltrada){
            console.log('Ganaste!!! ...');
        }
        const { letra } = await inquirer.prompt(pregunta);
        const indice = buscar(letra, palabraArrayFiltrada);
        palabraGenerada[indice] = letra;
    }

    console.log('Perdiste!!! No adivinaste la palabra ...\n');
    console.log(`La palabra era ${palabraArrayFiltrada}`);
        
}

const read = () => {

    const contenido = fs.readFileSync('./archivos/palabras.txt', 'utf-8');
    const palabras = contenido.split('\n');

    const palabrasArray = palabras.filter(item => item != '\r');
    const palabrasArray2 = palabrasArray.filter(item => item != '');

    const index = Math.floor(Math.random()*(palabrasArray2.length + 1));

    return palabrasArray2[index];
    
}

const buscar = (letra, arreglo = []) => {
    if (arreglo.indexOf(letra) == -1){
        console.log('La Letra no esta en la palabra ...\n'.red);
    } else {
        console.log(`La letra ${letra} está en la posicion ${arreglo.indexOf(letra)+1}\n`.green);

        return arreglo.indexOf(letra);
    }
}

const palabraBD = read();

menu(palabraBD);








